test/latlonbound_test_utilities.py

```diff
@@ -0,0 +1,386 @@
+"""
+This file stores the necessary helper functions for the latlonbox test cases.
+"""
+import copy
+
+import numpy as np
+
+import uxarray.helpers
+
+
+def max_latitude_rad(v1, v2):
+    """Quantitative method to find the maximum latitude between in a great circle arc
+
+    Parameters
+    ----------
+    v1: float array
+        The first endpoint of the great circle arc [lon, lat] in degree east.
+    v2: float array
+        The second endpoint of the great circle arc [lon, lat] in degree east.
+
+    Returns
+    -------
+    float
+        maximum latitude in radian
+    """
+
+    # Find the parametrized equation for the great circle passing through v1 and v2
+    err_tolerance = 1.0e-15
```

Comment:
What do you think about adding tolerance as an optional input to the function?

Reply:
The error tolerance here is not for the user input but for the python floating point precision and machine epsilon issues.

In other words, when we're doing the floating point calculation in python, we cannot get the exact same results every time (even though the data and the function are the same), we can at most achieve a certain level of precision genrerally. Therefore, after discussing with @paullric , we decided to make 1.0e-12 the precision level of all our calculations (When users use our function, they can expect the results to be at most 12 decimal precise).

The reason why I use 1.0e-15 here is just for precaution.